test/replication-luatest/gh_9918_synchro_queue_additional_info_test.lua

```diff
@@ -0,0 +1,102 @@
+local fiber = require('fiber')
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_each(function(cg)
+    cg.replica_set = replica_set:new{}
+    local box_cfg = {
+        replication = {
+            server.build_listen_uri('server1', cg.replica_set.id),
+            server.build_listen_uri('server2', cg.replica_set.id),
+        },
+        replication_timeout = 0.1,
+    }
+    cg.leader = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = box_cfg,
+    }
+    cg.replica = cg.replica_set:build_and_add_server{
+        alias = 'server2',
+        box_cfg = box_cfg,
+    }
+    cg.replica_set:start()
+    cg.leader:exec(function()
+        box.ctl.promote()
+        local s = box.schema.space.create('s', {is_sync = true})
+        s:create_index('pk')
+    end)
+    cg.replica_set:wait_for_fullmesh()
+    cg.leader:wait_for_downstream_to(cg.replica)
+end)
+
+g.after_each(function(cg)
+    cg.replica_set:drop()
+end)
+
+-- Test that new `age` field of `box.info.synchro.queue` works correctly.
+g.test_age_field = function(cg)
+    -- The synchronous queue is originally empty.
+    cg.leader:exec(function()
+        t.assert_equals(box.info.synchro.queue.age, 0)
+    end)
+    cg.replica:exec(function()
+        t.assert_equals(box.info.synchro.queue.age, 0)
+    end)
+
+    -- Ensure that the synchronous transactions stay in the queue for a while.
+    cg.leader:update_box_cfg{replication_synchro_quorum = 3,
```

Comment:
You don't revert box.cfg back (`replication_synchro_timeout`). Let's either put it in `before_all` trigger or properly revert the state of a leader at the end of a test.

Reply:
Moved to `before_all`.